### PR TITLE
fe2 android Some first communication 

### PR DESCRIPTION
**Pull request by zoemarin**
I start to create a dummie transaction

### CODE REVIEW

**Comment by jbsv** [comment]
It's hard to see what part of this PR is about communication. This PR is very similar to the 903. Maybe you could branch off the previous branch (see PR 903), and create a PR that will merge into this previous branch. This way you would have only the diff between the 2 branches.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashMain.java [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    private DigitalCashViewModel mViewModel;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.digital_cash_main_activity);
+        mViewModel = obtainViewModel(this);
+
+        // When we launch the social media from a lao, it directly sets its id and name
```
Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashMain.java:45]: Maybe create a variable for getIntent().getExtras()?

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    this.keyManager = keyManager;
+    disposables = new CompositeDisposable();
+
+    this.mLAOs =
+        LiveDataReactiveStreams.fromPublisher(
+            this.laoRepository.getAllLaos().toFlowable(BackpressureStrategy.BUFFER));
+  }
+
+  /*
+   * Getters for MutableLiveData instances declared above
```
Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java:100]: Maybe add comment for events? at least which fragment is calling them? It becomes a mess to read and understand when they accumulate :)

Comment (ahominal) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java:100]: Maybe add comment for events? at least which fragment is calling them? It becomes a mess to read and understand when they accumulate :)

**Review by ahominal — COMMENTED** [review]
Great work !

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Address.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.github.dedis.popstellar.model.objects;
+
+public class Address {
+
+    //Represent the address to which if a send and receiver of a transaction
+    // is actually the public key of some user I think
+    //String sender_address;
+    //String receiver_address;
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Address.java:8]: We should not have commented code

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java [hunk 4]
```diff
@@ -38,9 +40,13 @@
   private Map<String, RollCall> rollCalls;
   private Map<String, Election> elections;
   private Map<MessageID, Chirp> allChirps;
+  private Map<MessageID, DummyCoin> allDummyCoins;
   private Map<PublicKey, List<MessageID>> chirpsByUser;
+  private Map<PublicKey, List<MessageID>> dummyCoinByUser;
   private final Map<MessageID, ElectInstance> messageIdToElectInstance;
   private final Map<PublicKey, ConsensusNode> keyToNode;
+  // TODO some transaction
+  // Map <Public Key, Transaction>
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java:49]: Same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java [hunk 5]
```diff
@@ -68,7 +69,8 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
     }
     setupHomeButton();
     setupIdentityButton();
-    setupSocialMediaButton();
+    setupDigitalCashButton();
+    //setupSocialMediaButton();
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:73]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java [hunk 6]
```diff
@@ -85,7 +87,9 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
     // Subscribe to "open identity" event
     setupIdentityFragment();
     // Subscribe to " open social media " event
-    setupSocialMediaActivity();
+    // TODO: CHANGE Social MEDIA
+    // setupSocialMediaActivity();
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailActivity.java:91]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashSendFragment.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+  private void sendNewDummyCoin() {
+    //TODO: should be some check LAO
+    // send some coin
+    // make a toast appear
+    // if (mDigitalCashViewModel.getLaoId().getValue() == null) {
+    // Toast.makeText(
+    //        requireContext().getApplicationContext(), R.string.error_no_lao, Toast.LENGTH_LONG)
+    //       .show();
+    // } else {
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashSendFragment.java:86]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+    Toast.makeText(context, addTransaction.toString(), Toast.LENGTH_LONG).show();
+
+    Lao lao = getCurrentLao();
+    if (lao == null) {
+      Log.e(TAG, LAO_FAILURE_MESSAGE);
+      Toast.makeText(context, "FAIL DUMMY", Toast.LENGTH_LONG).show();
+      return;
+    }
+
+    //AddDummyTransaction addTransaction = new AddDummyTransaction(amount,sender_address,receiver_address);
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashViewModel.java:208]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 9]
```diff
@@ -64,7 +65,8 @@ protected void onCreate(Bundle savedInstanceState) {
     setupLaunchButton();
     setupConnectButton();
     setupWalletButton();
-    setupSocialMediaButton();
+    // setupSocialMediaButton();
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:68]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 10]
```diff
@@ -192,16 +195,30 @@ private void subscribeWalletEvents() {
...
+      //  .getOpenSocialMediaEvent()
+        //.observe(
+          //  this,
+            //booleanEvent -> {
+              //Boolean event = booleanEvent.getContentIfNotHandled();
+              //if (event != null) {
+                //setupSocialMediaActivity();
+              //}
+            //});
+  //}
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:210]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 11]
```diff
@@ -257,9 +274,14 @@ public void setupWalletButton() {
     walletButton.setOnClickListener(v -> mViewModel.openWallet());
   }
 
-  public void setupSocialMediaButton() {
-    Button socialMediaButton = findViewById(R.id.tab_social_media);
-    socialMediaButton.setOnClickListener(v -> mViewModel.openSocialMedia());
+  // public void setupSocialMediaButton() {
+  //  Button socialMediaButton = findViewById(R.id.tab_social_media);
+  //  socialMediaButton.setOnClickListener(v -> mViewModel.openSocialMedia());
+  // }
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:280]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java [hunk 12]
```diff
@@ -69,8 +69,10 @@
   private final MutableLiveData<SingleEvent<Boolean>> mOpenSeedEvent = new MutableLiveData<>();
   private final MutableLiveData<SingleEvent<String>> mOpenLaoWalletEvent = new MutableLiveData<>();
   private final MutableLiveData<SingleEvent<Boolean>> mOpenSettingsEvent = new MutableLiveData<>();
-  private final MutableLiveData<SingleEvent<Boolean>> mOpenSocialMediaEvent =
-      new MutableLiveData<>();
+  //private final MutableLiveData<SingleEvent<Boolean>> mOpenSocialMediaEvent =
+      //new MutableLiveData<>();
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java:73]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java [hunk 13]
```diff
@@ -285,8 +287,12 @@ public Boolean isWalletSetUp() {
     return mOpenSettingsEvent;
   }
 
-  public LiveData<SingleEvent<Boolean>> getOpenSocialMediaEvent() {
-    return mOpenSocialMediaEvent;
+  //public LiveData<SingleEvent<Boolean>> getOpenSocialMediaEvent() {
+    //return mOpenSocialMediaEvent;
+  //}
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java:292]: same as above

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java [hunk 14]
```diff
@@ -339,8 +345,12 @@ public void openSettings() {
     mOpenSettingsEvent.setValue(new SingleEvent<>(true));
   }
 
-  public void openSocialMedia() {
-    mOpenSocialMediaEvent.setValue(new SingleEvent<>(true));
+  //public void openSocialMedia() {
+    //mOpenSocialMediaEvent.setValue(new SingleEvent<>(true));
+  //}
```
Comment (Tesa27) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java:350]: same as above

**Review by Tesa27 — CHANGES_REQUESTED** [review]
Lot of commented code is present

**Pull request closed without merging**